annotation/tests/test_tasks_services.py

```diff
@@ -585,3 +636,1697 @@ def test_filter_tasks_db_no_files_or_jobs(
+    ) as mock_get_file_names_by_request, patch(
+        "annotation.tasks.services.lru_file_id_to_file_name_cache"
+    ) as mock_cache:
+
+        result = get_file_names_by_file_ids(file_ids, tenant, token)
+
+        mock_get_from_cache.assert_called_once_with(
+            objs_ids=file_ids, cache_used=mock_cache
+        )
+
+        mock_get_file_names_by_request.assert_called_once_with(
+            file_ids=list(not_cached_file_ids), tenant=tenant, token=token
+        )
+
+        mock_cache.update.assert_called_once_with(file_names_from_requests)
+
+        expected_result = {**cached_file_names, **file_names_from_requests}
+
+        assert result == expected_result
+
+
+@pytest.fixture
+def mock_metric():
+    metric = MagicMock()
+    metric.task_from = datetime(2024, 1, 1)
+    metric.task_to = datetime(2024, 2, 1)
+    metric.agreement_metric = True
+    yield metric
+
+
+@pytest.fixture
+def mock_stats(mock_task: ManualAnnotationTask, mock_metric: Mock):
+    stat1 = MagicMock()
+    stat1.task = mock_task
+    stat1.task_id = 1
+    stat1.created = datetime(2024, 1, 1, 12, 0, 0)
+    stat1.updated = datetime(2024, 1, 2, 12, 0, 0)
+    stat1.task.user_id = 2
+    stat1.task.file_id = 3
+    stat1.task.pages = [1, 2, 3]
+    stat1.task.status.value = "completed"
+    stat1.task.agreement_metrics = [mock_metric]
+
+    stat2 = MagicMock()
+    stat2.task = mock_task
+    stat2.task_id = 2
+    stat2.created = datetime(2024, 1, 3, 12, 0, 0)
+    stat2.updated = datetime(2024, 1, 4, 12, 0, 0)
+    stat2.task.user_id = 2
+    stat2.task.file_id = 3
+    stat2.task.pages = [1, 2, 3]
+    stat2.task.status.value = "completed"
+    stat2.task.agreement_metrics = [mock_metric]
+
+    stat3 = MagicMock()
+    stat3.task = mock_task
+    stat3.task_id = 3
+    stat3.created = datetime(2024, 1, 5, 12, 0, 0)
+    stat3.updated = datetime(2024, 1, 6, 12, 0, 0)
+    stat3.task.user_id = 2
+    stat3.task.file_id = 3
+    stat3.task.pages = [1, 2, 3]
+    stat3.task.status.value = "completed"
+    stat3.task.agreement_metrics = [mock_metric]
+
+    yield [stat1, stat2, stat3]
+
+
+@pytest.fixture
+def mock_db(mock_stats):
+    mock_db = MagicMock()
+    mock_query = MagicMock()
+    mock_query.filter.return_value.filter.return_value.all.return_value = (
+        mock_stats
+    )
+    mock_db.query.return_value = mock_query
+
+    return mock_db
+
+
+@pytest.mark.parametrize(
+    ("schema", "expected_filename"),
+    (
+        (
+            MagicMock(
+                user_ids=[2],
+                date_from=datetime(2024, 1, 1),
+                date_to=datetime(2024, 1, 6),
+            ),
+            f"annotator_stats_export_{datetime.now().strftime('%Y%m%d')}.csv",
+        ),
+        (
+            MagicMock(
+                user_ids=[999],
+                date_from=datetime(2024, 1, 1),
+                date_to=datetime(2024, 1, 6),
+            ),
+            f"annotator_stats_export_{datetime.now().strftime('%Y%m%d')}.csv",
```

Comment:
it doesn't change - move it from here. additionally, you rely on the current datetime, but what if you run tests at 23:59:59.999999 and during execution the next day begins? it will fail, so you should patch the time method

Reply:
remember - tests must not depend on any environment